### DISCUSSION TITLE
Triggering the windshield wiper through ONVIF

### DISCUSSION THREAD

**Discussion by xtrdustin** [AxisCommunications/discussions#139] [General] [answered]
We'd like to make it possible for users of our software to trigger the windshield wiper on the Axis camera. Is this possible through ONVIF, or would we need to use the [VAPIX API for it](https://www.axis.com/vapix-library/subjects/t10175981/section/t10164188/display?section=t10164188-t10164187)?

Thanks in advance!

**Comment by mattias-kindborg-at-work** [comment, marked as answer]
Hi @xtrdustin.

I think you should be able to send `GetServiceCapabilities` to the camera, to see if the camera support wipers or not:

```xml
<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:tds="http://www.onvif.org/ver10/device/wsdl" xmlns:tt="http://www.onvif.org/ver10/schema">
  <soap:Body>
    <tds:GetServiceCapabilities />
  </soap:Body>
</soap:Envelope>
```

Given the camera support it, the following examples would send the ONVIF command to trigger the wiper:

```xml
<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:tds="http://www.onvif.org/ver10/device/wsdl" xmlns:tt="http://www.onvif.org/ver10/schema">
  <soap:Body>
    <tds:SendAuxiliaryCommand>
      <tds:AuxiliaryCommand>tt:Wiper|On</tds:AuxiliaryCommand>
    </tds:SendAuxiliaryCommand>
  </soap:Body>
</soap:Envelope>
```

**Reply by xtrdustin** [reply]
Thanks, that worked! Specifically, I had to call the methods on the Device service (as opposed to the PTZ or Media services). There were also commands that turn the camera's IR mode on/off (or back to auto):

```
   misc=org.onvif.ver10.device.wsdl.MiscCapabilities@182e4365[
      auxiliaryCommands=[
         tt:IRLamp|On,
          tt:IRLamp|Off,
          tt:IRLamp|Auto,
          tt:Wiper|On,
          tt:Wiper|Off
      ],
```

**Reply by xtrdustin** [reply]
Hi @mattias-kindborg-at-work,

As a follow-up to this thread, do you know if there's a way to get the wiper status (either whether it's currently on, or how much longer is left in the cycle), through ONVIF? 

Thanks,
Dustin
